src/libs/installer/packagemanagercore.cpp

```diff
@@ -3063,6 +3063,39 @@ bool PackageManagerCore::fetchUpdaterPackages(const PackagesList &remotes, const
 
             std::sort(d->m_updaterComponents.begin(), d->m_updaterComponents.end(),
                 Component::SortingPriorityGreaterThan());
+
+            auto dependencySort = [](QList<QInstaller::Component *> components) {
+                QList<QInstaller::Component *> out;
+                QSet<QString> outNames;
+                QSet<QString> allNames;
+
+                // std::transform cannot insert into QSet easily and std::set is inconvenient to use
+                for (Component *component : components)
+                    allNames.insert(component->name());
+
+                auto hasNoPendingDeps = [&](QInstaller::Component *component) {
+                    const QSet<QString> requiredNames = parseNames(component->dependencies()).toSet();
+                    return !requiredNames.intersects(allNames - outNames);
```

Comment:
I'm probably missing something, but... Isn't the "allNames - outNames" unnecessary here, as the names are removed from allNames at the same time when they are added to outNames, on lines 3090 and 3091 below? Would just "allNames" provide the same result?

Reply:
Yeah two ideas mixed together.. Refactored.